src/components/volumeControl/VolumeControl.js

```diff
@@ -1,46 +1,46 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { withStyles } from '@material-ui/core/styles';
 import IconButton from '@material-ui/core/IconButton';
 import VolumeMuteIcon from '@material-ui/icons/VolumeMute';
 import VolumeUpIcon from '@material-ui/icons/VolumeUp';
+// It doesn't uss Css in JS because we needed to use a pseudo selector
+import './VolumeControl.css';
 
-const styles = {
-  root: {
-    width: 300,
-    display: 'flex'
-  },
-  slider: {
-    padding: '22px 0px'
-  },
-  blue: {
-    backgroundColor: '#00acc1',
-    margin: 0
-  },
-  track: {
-    margin: 0
+class VolumeControl extends React.Component {
+  state = {
+    value: 0
+  };
+
+  onChange = event => {
+    const value = event.target.value;
+    this.setState({ value: value });
+  };
+
+  onDragEnd = () => {
+    this.props.onVolumeChange(this.state.value);
+  };
+
+  componentDidMount() {
+    this.setState({ value: this.props.volume });
```

Comment:
This should either be `static getDerivedStateFromProps` or in the `constructor`.

Reply:
Thanks! Let me check that